Extension/src/LanguageServer/configurations.ts

```diff
@@ -386,6 +396,52 @@ export class CppProperties {
         }
     }
 
+    private nodeAddonMap: {[dependency: string]: string} = {
+        "nan": "node --no-warnings -e \"require('nan')\"",
+        "node-addon-api": "node --no-warnings -p \"require('node-addon-api').include\""
+    };
+
+    private async addNodeAddonIncludeLocations(rootPath: string): Promise<void> {
+        try {
+
+            const settings: CppSettings = new CppSettings(this.rootUri);
+            if (settings.addNodeAddonIncludePaths) {
+                const package_json: any = JSON.parse(await pfs.readFile(path.join(rootPath, "package.json"), "utf8"));
```

Comment:
It seems like it would be better to check for this file's existence first?

Reply:
yes, i should or the user will get a message about an unhandled promise rejection.

i will change this logic to better capture errors and issue better messages. should i use `vscode.window.showErrorMessage()` to display a message if there is a problem? it doesn't seem like using console.log is optimal.

Reply:
For serious errors that users should always see, showErrorMessage should be used, the console.log may only get seen if the user looks at the logs or does a Toggle Developer Console. For this case, it seems like it should just silently skip the processing, since the user would just be opening a non-nodejs project.